lib/created_message_queue_handler.ts

```diff
@@ -112,34 +111,73 @@ type ContextWithBindings = t.TypeOf<typeof ContextWithBindings> & IContext;
  * @param maybeProfile      the recipient's profile (or none)
  * @param defaultAddresses  default addresses (one per channel)
  */
-function getEmailNotification(
+function tryGetDestinationEmail(
   maybeProfile: Option<RetrievedProfile>,
```

Comment:
pls add a TODO here, we need to refactor this method and the next one, methods shouldn't expect `Option` as parameters, this is almost always a code smell that indicated a leakage of control flow from the caller to the callee, it should be responsibility of the caller to check the content of an `Option`

Reply:
are you suggesting to pass an undefined value instead ? (ie. defaultAddress)